### PR TITLE
tests(e2e): Environment configs E2Es

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Based on #4343, so please only look at the last commit, will be rebased once that gets merged.

Adds E2Es for #3981.

To be refactored once we agree on how to handle shared setups for groups of features.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: test/e2e/funcs/feature.go [hunk 1]
```diff
@@ -286,6 +288,11 @@ func ResourcesHaveFieldValueWithin(d time.Duration, dir, pattern, path string, w
 			return ctx
 		}
 
+		if count.Load() == 0 {
```
Comment (lsviben) [inline test/e2e/funcs/feature.go:291]: Didnt dig into it, but shouldn't the `wait.For(..` return an err if no resources matched the pattern in the given time?

File: test/e2e/install_test.go [hunk 2]
```diff
@@ -56,6 +58,7 @@ func TestCrossplaneLifecycle(t *testing.T) {
 			WithLabel(LabelSize, LabelSizeSmall).
 			WithLabel(LabelModifyCrossplaneInstallation, LabelModifyCrossplaneInstallationTrue).
 			WithLabel(config.LabelTestSuite, config.TestSuiteDefault).
+			WithLabel(config.LabelTestSuite, TestSuiteLifecycle).
```
Comment (lsviben) [inline test/e2e/install_test.go:61]:
```suggestion
			WithLabel(config.LabelTestSuite, TestSuiteLifecycle).
```

**Review by lsviben — APPROVED** [review]
Thanks @phisco, LGTM, added a non-blocking question and a small fix suggestion

File: test/e2e/funcs/feature.go (hunk 1 above)
Comment (phisco) [reply test/e2e/funcs/feature.go:291]: Yes, but not if you asked it to wait for no resources at all in the first place 😉

File: test/e2e/install_test.go (hunk 2 above)
Comment (phisco) [reply test/e2e/install_test.go:61]: This was intentional, features can have multiple values for each label. This allows the environment config suite to run the upgrade tests and not the whole default suite.

**Pull request merged**